pymeasure/instruments/hp/hp8753e.py

```diff
@@ -0,0 +1,394 @@
+        for i in range(5):
+            try:
+                averaging_enabled = self.averaging_enabled
+                break
+            except VisaIOError as e:
+                sleep(0.2)
+                if e.abbreviation != "VI_ERROR_TMO":
+                    raise e
+                if i == 4:
+                    raise e
+        if averaging_enabled:
+            self.write(f"NUMG{self.averages}")
+        else:
+            self.write("SING")
+
+    def scan_continuous(self):
+        """Initiates a continuous scan"""
+        self.write("CONT")
+        self.adapter.flush_read_buffer()
+
+    @property
+    def frequencies(self):
+        """Returns a list of frequencies from the last scan"""
+        return np.linspace(self.start_frequency, self.stop_frequency, num=self.scan_points)
+
+    @property
+    def data_complex(self):
+        """Returns the complex power from the last scan"""
+        # TODO: Implement binary transfer instead of ASCII
+
+        # get number of points
+        points = self.scan_points
+
+        # TODO get data format
+        self.write("FORM4")
+
+        self.adapter.flush_read_buffer()
+
+        if isinstance(self.adapter, PrologixAdapter):
+            temp_data = []
+
+            # set adapter to auto
+            auto_state = self.adapter.auto
+            self.adapter.auto = 1
+            start = now()
+            self.write("OUTPDATA")
+
+            # if form4 is the data transfer method
+            while len(temp_data) < (points):
+                if self.adapter.connection.bytes_in_buffer >= 50:
+                    temp_data.append(self.read_bytes(50).decode("utf-8").strip("\n").split(","))
+                else:
+                    # try not to overwhelm the prologix adapter
+                    sleep(0.0001)
+
+                # break the loop if the sweep gets interrupted
+                time_elapsed = now() - start
+                if time_elapsed > points * 0.006:
+                    if self.adapter.connection.bytes_in_buffer > 0:
+                        self.adapter.flush_read_buffer()
+                        sleep(0.05)
+                    raise Exception(f"Failed to read data. Data transfer method timed out")
+
+            # process string data into complex numbers
+            preformat_data = [float(point[0]) + float(point[1]) * 1j for point in temp_data]
+            data = np.array(preformat_data)
+
+            # return adapter to previous auto state for prologix adapter
+            self.adapter.auto = auto_state
+
+            # data_complex = data[:, 0] + 1j * data[:, 1]
+            return data
+        else:
+            raise NotImplementedError("Function data_complex only written for PrologixAdapter")
+
+
+if __name__ == "__main__":
+    adapter = PrologixAdapter("ASRL4::INSTR", address=16, visa_library="@py")
+    # adapter = PrologixAdapter("ASRL4::INSTR", address=16)
+
+    vna = HP8753E(adapter)
+
+    vna.adapter.auto = 1
+    print(vna.id)
```

Comment:
As you said, we do not use this code block, as everyone has his own configuration.

Reply:
Thank you so much for the quick feedback! I'll work on implementing your feedback in addition to my other to-do items.